src/app/modules/item/task/task-xd-pr.ts

```diff
@@ -0,0 +1,303 @@
+    this.chan.bind('platform.showView', (view : TaskView) => platform.showView(view));
+    this.chan.bind('platform.askHint', (hintToken : string) => platform.askHint(hintToken));
+    this.chan.bind('platform.updateDisplay', (data : TaskDisplayData) => platform.updateDisplay(data));
+    this.chan.bind('platform.openUrl', (url : string) => platform.openUrl(url));
+    this.chan.bind('platform.log', (data : TaskLog) => platform.log(data));
+
+    // Legacy calls
+    this.chan.bind('platform.updateHeight', (height : number) => platform.updateDisplay({ height: height }));
+  }
+
+  /**
+   * Task API functions
+   */
+  load(views : Object): Observable<void> {
+    return this.chan.call({
+      method: "task.load",
+      params: views,
+    });
+  }
+
+  unload() : Observable<void> {
+    return this.chan.call({
+      method: "task.unload",
+      timeout: 2000,
+    });
+  }
+
+  getHeight() : Observable<number> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getHeight",
+      timeout: 500,
+    });
+  }
+
+  updateToken(token : string) : Observable<void> {
+    return this.chan.call({
+      method: "task.updateToken",
+      params: token,
+      timeout: 10000,
+    });
+  }
+
+  getMetaData() : Observable<TaskMetaData> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getMetaData",
+      timeout: 2000
+    });
+  }
+
+  getAnswer() : Observable<string> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getAnswer",
+      timeout: 2000
+    });
+  }
+
+  reloadAnswer(answer : string) : Observable<void> {
+    return this.chan.call({
+      method: "task.reloadAnswer",
+      params: answer,
+      timeout: 2000
+    });
+  }
+
+  getState() : Observable<string> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getState",
+      timeout: 2000
+    });
+  }
+
+  reloadState(state : string) : Observable<void> {
+    return this.chan.call({
+      method: "task.reloadState",
+      params: state,
+      timeout: 2000
+    });
+  }
+
+  getViews() : Observable<TaskViews> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getViews",
+      timeout: 2000
+    });
+  }
+
+  showViews(views : Object) : Observable<void> {
+    return this.chan.call({
+      method: "task.showViews",
+      params: views,
+      timeout: 2000
+    });
+  }
+
+  gradeAnswer(answer : string, answerToken : string) : Observable<TaskGrade> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.gradeAnswer",
+      params: [ answer, answerToken ],
+      timeout: 40000
+    });
+  }
+
+  getResources() : Observable<TaskResources> {
+    // TODO: validator
+    return this.chan.call({
+      method: "task.getResources",
+      params: [],
+      timeout: 2000
+    });
+  }
+}
+
+/*
+ * Platform object definition, created from a Task object (see below)
+ */
+
+export class Platform {
```

Comment:
In the API it's called Platform, and in this file the name is used for instance in the method names bound in Task.setPlatform (`'platform.validate'`) which can't be changed because of the API, so changing the name here would probably be confusing.

Reply:
Renamed to TaskListener. We could see about renaming other functions too if their name is not clear.